pipeline/delivlib.ts

```diff
@@ -81,11 +78,66 @@ export class DelivLibPipelineStack extends Stack {
   }
 }
 
+export class EcrMirrorStack extends Stack {
+  public readonly superchainRepo: ecr.IRepository;
+
+  constructor(scope: App, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const superchainSource = delivlib.MirrorSource.fromDockerHub(DOCKERHUB_SUPERCHAIN);
+
+    const ecrMirror = new delivlib.EcrMirror(this, 'Default', {
+      dockerHubCreds: {
+        secret: secret.Secret.fromSecretArn(this, 'DockerHubCreds', 'arn:aws:secretsmanager:us-east-1:712950704752:secret:dockerhub/ReadOnly-VXZo5Z'),
+        usernameKey: 'username',
+        passwordKey: 'password',
+      },
+      images: [
+        superchainSource,
+      ],
+    });
+
+    const repo = ecrMirror.ecrRepository(superchainSource);
+    if (!repo) {
+      throw new Error('Cannot find ECR mirror repository for "jsii/superchain"');
+    }
+    this.superchainRepo = repo;
+  }
+}
+
+export class EcrMirrorAspect implements IAspect {
+  constructor(private readonly superchainRepo: ecr.IRepository) {}
+
+  public visit(construct: IConstruct) {
```

Comment:
This is cool, from a technical perspective, but actually exposes a problem with non configurable build images in delivlib. Is it a big effort to make them configurable and just pass the ECR image?

Reply:
There's too many places this occurs - autobuild, autoPR, mergeback, regular builds. Trying to pull them all out from underneath the layers isn't trivial.

Isn't this a good use case for an aspect? I'm even wondering if we should export this out of delivlib.

Reply:
I had always considered Aspects to be more of an augmentation/validation, rather than mutation. Personally I find this a little unexpected.

But thats fine, we can get back to it at some point.